vispy/scene/canvas.py

```diff
@@ -207,6 +207,7 @@ def draw_visual(self, visual, event=None, viewport=None):
             Optionally specifies the viewport to use. If None, the entire
             physical size is used.
         """
+        self.set_current()
```

Comment:
I seem to remember reading at one point that context switches can be slow. Can you check to make sure this doesn't hurt performance (see how non-vsync'ed framerate drops with and without this line even if nothing is drawn)?